src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java

```diff
@@ -439,53 +443,43 @@ protected JsonToken _decodeScalar(ScalarEvent scalar)
         String typeTag = scalar.getTag();
         final int len = value.length();
 
-        if (typeTag == null) { // no, implicit
-            // We only try to parse the string value if it is in the plain flow style.
-            // The API for ScalarEvent.getStyle() might be read as a null being returned
-            // in the plain flow style, but debugging shows the null-byte character, so
-            // we support both.
-            Character style = scalar.getStyle();
-
-            if ((style == null || style == '\u0000') && len > 0) {
-                char c = value.charAt(0);
-                switch (c) {
-                case 'n':
-                    if ("null".equals(value)) {
-                        return JsonToken.VALUE_NULL;
-                    }
-                case '0':
-                case '1':
-                case '2':
-                case '3':
-                case '4':
-                case '5':
-                case '6':
-                case '7':
-                case '8':
-                case '9':
-                case '+':
-                case '-':
-                case '.':
-                    JsonToken t = _decodeNumberScalar(value, len);
-                    if (t != null) {
-                        return t;
-                    }
-                }
+        if (typeTag == null || typeTag.equals("!")) { // no, implicit
+            Tag nodeTag = _yamlResolver.resolve(NodeId.scalar, value, scalar.getImplicit().canOmitTagInPlainScalar());
+            if (nodeTag == Tag.STR) {
+                return (_currToken = JsonToken.VALUE_STRING);
+            } else if (nodeTag == Tag.INT) {
+                return _decodeNumberScalar(value, len);
+            } else if (nodeTag == Tag.FLOAT) {
+                _numTypesValid = 0;
+                return JsonToken.VALUE_NUMBER_FLOAT;
+            } else if (nodeTag == Tag.BOOL) {
                 Boolean B = _matchYAMLBoolean(value, len);
                 if (B != null) {
-                    return B.booleanValue() ? JsonToken.VALUE_TRUE : JsonToken.VALUE_FALSE;
+                    return B ? JsonToken.VALUE_TRUE : JsonToken.VALUE_FALSE;
                 }
+            } else if (nodeTag == Tag.NULL) {
+                return JsonToken.VALUE_NULL;
+            } else {
+                // what to do with timestamp and binary and merge etc.
+                return JsonToken.VALUE_STRING;
             }
         } else { // yes, got type tag
+            if (typeTag.startsWith("tag:yaml.org,2002:")) {
```

Comment:
You went ahead and merged. I think this is all in the YAML 1.1 spec; see http://yaml.org/spec/1.1/#id861700, see section 2.4, see 3.1.1. The tags like this are scattered all over the doc, and never in a list.